Port to the Meson buildsystem

hughsie opened the pull request.

hughsie commented:
@vathpela you can be very honest here -- if you want me to break this up a bit that's fine too.

vathpela commented:
So I definitely want at least the coverity stuff split out as another patch (though it'd be nice to keep it working...) and the examples don't seem to work for me... Tbh not very happy with how e.g. the fanalyzer workflow goes.

Also can we fix it so `ninja dist` builds the .spec file into the tar all correctly?

(CI roots are rebuilding with meson+ninja in them now but it takes a while and it's Zelda time now.)

hughsie commented:
> the coverity stuff split out as another patch

Quick sanity check, the old coverity failure happened *client side* rather than on the covscan servers, right?

	Coverity Build Capture (64-bit) version 2022.12.2 on Linux 6.7.9-200.fc39.x86_64 x86_64
	Internal version numbers: 126f4dac91 p-2022.12-push-76
	[9/9] Linking target src/eficonman
	Emitted 5 C/C++ compilation units (100%) successfully
	5 C/C++ compilation units (100%) are ready for analysis
	The cov-build utility completed successfully.

> and the examples don't seem to work for me

I've added some more guidance to `README.md` and removed the extra `build`s -- did that help?

> Also can we fix it so ninja dist builds the .spec file into the tar all correctly

I don't think we *can* -- it's generated at build time as it's using the overridable `@VERSION@` -- but if you `ninja` it -- it pops up in `build` -- so we can do

	rpmbuild -ba efibootmgr.spec
 	...
	Wrote: /home/hughsie/rpmbuild/SRPMS/efibootmgr-18-1.fc39.src.rpm
	Wrote: /home/hughsie/rpmbuild/RPMS/efibootmgr-debugsource-18-1.fc39.x86_64.rpm
	Wrote: /home/hughsie/rpmbuild/RPMS/efibootmgr-18-1.fc39.x86_64.rpm
	Wrote: /home/hughsie/rpmbuild/RPMS/efibootmgr-debuginfo-18-1.fc39.x86_64.rpm
	...
	+ exit 0
